### DISCUSSION TITLE
'allowEmptyValue' for query

### DISCUSSION THREAD

**Discussion by rockyoung** [fastapi/fastapi#11488] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Query

app = FastAPI()

@app.get("/foo")
async def foo(qk: str|None=Query(default='any'))
```

### Description

First of all, thanks a lot for such excellent work by all contributors!

I want the `qk` is an optional query parameter, and has the default value 'any' **ONLY** when it is omitted (or say None) in query, but not empty value like `/foo?otherkey=something&qk` or `/foo?qk=&otherkey=something`, i need to distinguish them from `/foo?noqk=something` which is the omitted situation.
It is ok as expected if i just test them using my own written cmd like curl, but when using the feature 'Try it out' in docs (the swagger ui), things become hard: empty input will always consider as the omitted case after pressing the 'Execute' button 😢.
Than i did some searching and read the document of both FastAPI and swagger, confirmed that current version of FastAPI can not support this kind of feature. swagger do have a [describing config for the parameter](https://swagger.io/docs/specification/describing-parameters) called 'allowEmptyValue', which will show a checkbox to allow user to send empty value if it is set to true. i set it up separately in `json_schema_extra` of `Query`  and  `swagger_ui_parameters` of `FastAPI`, neither of them successed, because both of them insert at the wrong place (checking by the 'openapi.json').
so do i missed something important that i havn't noticed yet? if no, is it reasonable to add a parameter to `Query` for this feature, like `allow_empty_send_in_swagger`?

### Operating System

Linux

### FastAPI Version

0.110.2

### Pydantic Version

2.7.1

### Python Version

3.10.13

**Comment by YuriiMotov** [comment, marked as answer]
I managed to make it work only with openapi customization.

```
from fastapi import FastAPI, Query
from fastapi.openapi.utils import get_openapi

app = FastAPI()

@app.get("/foo")
async def foo(qk: str = Query(default='any')) -> str:
    return qk


def custom_openapi():
    if app.openapi_schema:
        return app.openapi_schema
    openapi_schema = get_openapi(
        title=app.title,
        version=app.version,
        summary=app.summary,
        description=app.description,
        routes=app.routes,
    )
    for param in openapi_schema["paths"]["/foo"]["get"]["parameters"]:
        if param["name"] == "qk":
            param["schema"].pop("default")
            param.pop("required")
            param["allowEmptyValue"] = True
    app.openapi_schema = openapi_schema
    return app.openapi_schema


app.openapi = custom_openapi
```

**Reply by rockyoung** [reply]
Thank you for your reply 👍, it works like a charm, even though it doesn't look elegant 😿
